api/cmd/formation/handler/ec2.go

```diff
@@ -109,5 +132,104 @@ func EC2NatGatewayDelete(req Request) (string, map[string]string, error) {
 		NatGatewayId: aws.String(req.PhysicalResourceId),
 	})
 
+	// block for 2 minutes until it's deleted
+	// Fixes subsequent CF error on deleting Elastic IP:
+	//   API: ec2:disassociateAddress You do not have permission to access the specified resource.
+	for i := 0; i < 12; i++ {
+		resp, derr := EC2(req).DescribeNatGateways(&ec2.DescribeNatGatewaysInput{
+			NatGatewayIds: []*string{aws.String(req.PhysicalResourceId)},
+		})
+
+		if derr != nil {
+			fmt.Printf("EC2NatGatewayDelete error: %s\n", derr)
+
+			// if nat gateway not found, break
+			if ae, ok := derr.(awserr.Error); ok {
+				if ae.Code() == "InvalidParameterException" {
+					break
+				}
+			}
+		}
+
+		// if NAT gateway is deleted, break
+		if len(resp.NatGateways) == 1 {
+			n := resp.NatGateways[0]
+
+			if *n.State == "deleted" {
+				break
+			}
+		}
+
+		// sleep and retry
+		time.Sleep(10 * time.Second)
+	}
+
+	// return original DeleteNatGateway success / failure
+	return req.PhysicalResourceId, nil, err
+}
+
+func EC2RouteCreate(req Request) (string, map[string]string, error) {
```

Comment:
You're right it's not used. But for now we are copy/pasting the function def from the other handlers that do use this.